accession-commons-core/src/main/java/uk/ac/ebi/ampt2d/commons/accession/core/AccessioningService.java

```diff
@@ -27,18 +27,39 @@
 
 import java.util.List;
 
+/**
+ * Service for retrieval and modifications of object accessions.
+ *
+ * @param <MODEL> Type of the objects identified by the accessions
+ * @param <HASH> Type of the hash calculated based on the fields that uniquely identify an accessioned object
+ * @param <ACCESSION> Accession of the object that identify a particular model
+ */
 public interface AccessioningService<MODEL, HASH, ACCESSION> {
 
+    /**
+     * Find the accessions associated with a list of objects.
+     * Searches object's accession in the repository, and if it does not exist, new accession is generated and stored in repository
+     *
+     * @param messages List of messages to be accessioned
+     * @return Wrappers containing the objects and their associated accession and hash
```

Comment:
`List of wrapper objects containing the accessioned objects and their associated accessions and hashes`?

Reply:
Lets hear from others

Reply:
agree with @Zhicheng-Liu

Reply:
Agree with @Zhicheng-Liu's proposal, it is less ambiguous.